internal/controllers/topology/cluster/patches/inline/json_patch_generator.go

```diff
@@ -170,9 +171,15 @@ func matchesSelector(req *runtimehooksv1.GeneratePatchesRequestItem, templateVar
 				// If templateMDClass matches one of the configured MachineDeploymentClasses.
 				for _, mdClass := range selector.MatchResources.MachineDeploymentClass.Names {
 					// We have to quote mdClass as templateMDClassJSON is a JSON string (e.g. "default-worker").
-					if string(templateMDClassJSON.Raw) == strconv.Quote(mdClass) {
+					if mdClass == "*" || string(templateMDClassJSON.Raw) == strconv.Quote(mdClass) {
 						return true
 					}
+					regex, err := regexp.Compile(mdClass)
```

Comment:
can we simply test prefix suffix instead of compiling a regex for every md class?